### PR TITLE
fix(jenkins): additional process to avoid panic

### PR DESCRIPTION
**Pull request by warren830**
### Summary
According to issue#3757, user met the error which related to line 67 in subtask `enrichApiBuildWithStages`
I checked the code of that line `build.JobName = strings.TrimSpace(str[0])`, I can only assume the error was caused by `str[0]` which means len(str) = 0.
To avoid this panic, I tried to add some check to avoid panic by the following situations:
1. After splitting stage's buildName by '#', and the string list of that result has less than 2 elements, we can just ignore this build
2. If the last element of the string list above is not a number, which means we got an error after converting it to number, then we again ignore the build

### Does this close any open issues?
Relates to #3757 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/jenkins/e2e/raw_tables/_raw_jenkins_api_stages.csv [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-13583,"{""ConnectionId"":1}","{""_links"":{""self"":{""href"":""/job/Test%20Gitlab%20Sync/10/execution/node/6/wfapi/describe""}},""id"":""6"",""name"":""gitlabAutoSync"",""execNode"":"""",""status"":""SUCCESS"",""startTimeMillis"":1583981120136,""durationMillis"":86044,""pauseDurationMillis"":0}",https://jenkins.merico.cn/job/Test%20Gitlab%20Sync/10/wfapi/describe,"{""Number"": ""10"", ""JobName"": ""Test Gitlab Sync"", ""DisplayName"": ""Test Gitlab Sync #10""}",2022-08-11 09:15:52.210
+1,"{""ConnectionId"":1}","{""_links"":{""self"":{""href"":""/job/pipeline-test2/2/execution/node/6/wfapi/describe""}},""id"":""6"",""name"":""Hello"",""execNode"":"""",""status"":""SUCCESS"",""startTimeMillis"":1662651634424,""durationMillis"":79,""pauseDurationMillis"":0}","https://test.nddtf.com/job/pipeline-test2/2/wfapi/describe","{""Path"": """", ""Number"": ""2"", ""JobName"": ""pipeline-test2"", ""FullDisplayName"": ""pipeline-test2 #2""}","2022-09-09 08:39:47.763"
```
Comment (likyh) [inline plugins/jenkins/e2e/raw_tables/_raw_jenkins_api_stages.csv:2]: why "DisplayName" change to "FullDisplayName"

File: plugins/jenkins/e2e/raw_tables/_tool_jenkins_stages.csv [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+1,6,pipeline-test2 #2,Hello,,SUCCESS,1662651634424,79,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,6,
+1,6,pipeline-test2 #3,Hello,,SUCCESS,1662651649629,122,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,3,
+1,6,Test Gitlab Sync #10,gitlabAutoSync,,SUCCESS,1583981120136,86044,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13583,
+1,6,Test Gitlab Sync #12,gitlabAutoSync,,SUCCESS,1583981125465,83641,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13581,
+1,6,Test Gitlab Sync #3,gitlabAutoSync,,SUCCESS,1581076468986,14118,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13579,
+1,6,Test Gitlab Sync 11,gitlabAutoSync,,SUCCESS,1583981120136,86044,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13584,
+1,8,Pipeline expirement #5,scp-f/b,,FAILED,1572321694770,297,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13577,
+1,6,UItest10.90.0.66 #11,gitlabAutoSync,,SUCCESS,1583981120136,86044,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13584,
+1,6,UItest10.90.0.66 12,gitlabAutoSync,,SUCCESS,1583981120136,86044,0,,"{""ConnectionId"":1}",_raw_jenkins_api_stages,13584,
```
Comment (likyh) [inline plugins/jenkins/e2e/raw_tables/_tool_jenkins_stages.csv:16]: add some data in another connection_id

File: plugins/jenkins/e2e/builds_test.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ func TestJenkinsBuildsDataFlow(t *testing.T) {
 	dataflowTester.FlushTabler(&models.JenkinsBuild{})
 	dataflowTester.FlushTabler(&models.JenkinsBuildCommit{})
 	dataflowTester.FlushTabler(&models.JenkinsStage{})
+	dataflowTester.ImportCsvIntoTabler("./raw_tables/_tool_jenkins_stages.csv", &models.JenkinsStage{})
 
 	dataflowTester.Subtask(tasks.ExtractApiBuildsMeta, taskData)
```
Comment (likyh) [inline plugins/jenkins/e2e/builds_test.go:51]: add some test data in e2e test to cover the logic about `JenkinsBuild. HasStages`

File: plugins/jenkins/e2e/raw_tables/_raw_jenkins_api_stages.csv (hunk 1 above)
Comment (warren830) [reply plugins/jenkins/e2e/raw_tables/_raw_jenkins_api_stages.csv:2]: Because jenkins actually can use fullDisplayName as pk

**Review by likyh — APPROVED** [review]
CSV is so difficult to understand. Have a try.

**Pull request merged**